passes/sat/sim.cc

```diff
@@ -287,16 +287,18 @@ struct SimInstance
 
 			if (mod != nullptr) {
 				dirty_children.insert(new SimInstance(shared, scope + "." + RTLIL::unescape_id(cell->name), mod, cell, this));
-			}
-
-			for (auto &port : cell->connections()) {
+			} else {
+				for (auto &port : cell->connections())
```

Comment:
What's the reasoning for moving this into an else block?  Is marking the bits dirty redundant if the child is already marked dirty?